src/hotspot/share/cds/cdsConfig.cpp

```diff
@@ -546,6 +548,32 @@ bool CDSConfig::check_vm_args_consistency(bool patch_mod_javabase, bool mode_fla
   return true;
 }
 
+void CDSConfig::setup_compiler_args() {
+  // AOT profiles are supported only in the JEP 483 workflow.
+  bool can_dump_profiles = AOTClassLinking && new_aot_flags_used();
```

Comment:
Why check `new_aot_flags_used()`? As I understand `AOTClassLinking` should be enough in mainline. Am I missing something?

Reply:
@iklam, do I need it?